src/lm/arpa-lm-compiler.h

```diff
@@ -38,8 +38,8 @@ class ArpaLmCompiler : public ArpaFileParser {
   }
   ~ArpaLmCompiler();
 
-  const fst::StdVectorFst& Fst() const { return fst_; }
-  fst::StdVectorFst* MutableFst() { return &fst_; }
+  const fst::StdVectorFst& Fst() const { CheckFstInvariants(); return fst_; }
+  fst::StdVectorFst* MutableFst() { CheckFstInvariants(); return &fst_; }
```

Comment:
This is cool, but I'd prefer if you put this check at the end of ReadComplete().  It gives fewer things to worry about.
And probably rename CheckFstInvariants() to Check(), and make it private-- just because I normally use that name for functions that do a check of some kind.

Reply:
Your suggestions are reasonable. I'll plan to address them tonight. By the way, CheckFstInvariants() was already private.